app/charts/map/map.tsx

```diff
@@ -204,17 +206,30 @@ export const MapComponent = () => {
   const geoJsonLayerId = useMemo(() => {
     return globalGeoJsonLayerId++;
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [areaLayer.getValue, areaLayer.getColor, mapStyle]);
+  }, [
+    areaLayer.getValue,
+    areaLayer.hierarchyLevel,
+    areaLayer.getColor,
+    mapStyle,
+  ]);
   const geoJsonLayer = useMemo(() => {
     if (!areaLayer.show) {
       return;
     }
-    const shapes = {
-      ...features.areaLayer?.shapes,
-      features: features.areaLayer?.shapes?.features.filter(
+    // Sort for smaller shapes to be over larger ones, to be able to use tooltip
+    const sortedFeatures = orderBy(
+      features.areaLayer?.shapes?.features.filter(
         ({ properties: { hierarchyLevel } }: GeoFeature) =>
           hierarchyLevel === areaLayer.hierarchyLevel
       ),
+      (feature: GeoFeature) => {
+        return area(feature);
+      },
```

Comment:
Nit: Maybe [d3.geoArea](https://github.com/d3/d3-geo#geoArea) could be used here so we don't need to install another package?